### ISSUE TITLE
promt

### ISSUE THREAD

**Issue by Wang-tiantian** [Wang-tiantian/test#27]
# kontext-dev生成编辑批跑

import torch
from diffusers import FluxKontextPipeline
from diffusers.utils import load_image

from app import prompt
seed = 88
torch.manual_seed(seed)
torch.cuda.manual_seed(seed)

pipe = FluxKontextPipeline.from_pretrained("./FLUX.1-Kontext-dev", torch_dtype=torch.bfloat16)
pipe.to("cuda")

prompt_list = [
    # 1. 测试复杂光影与材质（玻璃、丝绸、折射光）
    "A glass marble on a wrinkled silk cloth, refracting sunlight",

    # 2. 测试多主体互动与构图逻辑（猫狗下棋）
    "A cat and dog playing chess in a library",

    # 3. 测试超现实场景生成（反重力瀑布）
    "A floating island with a waterfall pouring into the sky",

    # 4. 测试高精度人脸与表情（避免畸形）
    "A close-up portrait of an old woman laughing, detailed wrinkles",

    # 5. 测试动态模糊与运动渲染
    "A racing car speeding past neon signs, motion blur",

    # 6. 测试文字生成能力（双语混合）
    "A chalkboard with 'Hello 你好' written in colorful chalk",

    # 7. 测试文化元素混合（赛博+武士）
    "A cyberpunk samurai standing in Tokyo rain",

    # 8. 测试微观细节与视角（昆虫视角）
    "A bee's eye view of a sunflower pollen field",

    # 9. 测试抽象艺术表现力
    "Blue and gold fluid art resembling a galaxy",

    # 10. 测试逻辑冲突兼容性（中世纪+摩托）
    "A medieval knight charging on a motorcycle"
]
edit_prompt_list = [
    # 原图1: "A glass marble on a wrinkled silk cloth, refracting sunlight"
    [
        "Replace the glass marble with a polished silver ball, keeping the silk and lighting",  # 材质替换
        "Change the sunlight to neon pink and blue lighting, dramatic reflections",            # 光效调整
        "Ultra-high detail close-up of the marble's refraction, 8K texture",                  # 超分辨率
        "Add tiny glowing runes floating around the marble, fantasy style",                   # 元素添加
        "Make the silk cloth wet with water droplets, hyper-realistic"                        # 物理状态修改
    ],
    # 原图2: "A cat and dog playing chess in a library"
    [
        "Make the cat and dog fighting over the chessboard, books falling",                    # 动作修改
        "Change the library to a sci-fi space station, keep chess game",                       # 背景替换
        "Add a parrot watching the game from a bookshelf, detailed feathers",                  # 新增主体
        "Convert the scene to oil painting style, visible brushstrokes",                       # 风格迁移
        "Replace chess pieces with miniature dinosaurs, playful mood"                          # 元素替换
    ],
    # 原图3: "A floating island with a waterfall pouring into the sky"
    [
        "Make the waterfall fall downward normally, keep floating island",                     # 逻辑修正
        "Convert the scene to 16-bit pixel art style, vibrant colors",                         # 风格化
        "Add ancient ruins and rope bridges on the island",                                    # 建筑扩展
        "Change the sky to a nebula with stars, cosmic atmosphere",                            # 环境特效
        "Make the island a giant turtle shell, fantasy creature"                               # 创意变形
    ],
    # 原图4: "A close-up portrait of an old woman laughing, detailed wrinkles"
    [
        "Make her look 30 years younger, keep expression and composition",                     # 年龄修改
        "Change her expression to angry, wrinkles more pronounced",                            # 表情调整
        "Transform the portrait into Van Gogh's brushstroke style",                            # 艺术风格
        "Add a glowing cybernetic eye, sci-fi augmentation",                                   # 局部增强
        "Make her wear a jeweled crown, royal portrait lighting"                               # 属性添加
    ],
    # 原图5: "A racing car speeding past neon signs, motion blur"
    [
        "Remove motion blur, show the car sharply with frozen sparks",                         # 动态效果
        "Change the scene to daytime, neon signs turned off",                                   # 时间切换
        "Replace the car with a futuristic motorcycle, keep speed effect",                     # 主体替换
        "Add a giant robot chasing the car, cinematic explosion",                              # 冲突场景
        "Convert to black and white vintage photo, grain texture"                               # 风格复古化
    ],
    # 原图6: "A chalkboard with 'Hello 你好' written in colorful chalk"
    [
        "Erase the text, write 'E=mc²' in messy chalk style",                                   # 文字修改
        "Change the chalkboard to a medieval stone wall, keep chalk texture",                  # 背景融合
        "Make the chalk letters pop out in 3D, realistic shadows",                              # 立体化
        "Add a ghostly hand writing the words, spooky atmosphere",                             # 超自然元素
        "Convert the text to glowing holographic projection"                                    # 科技感改造
    ],
    # 原图7: "A cyberpunk samurai standing in Tokyo rain"
    [
        "Replace the armor with a tailored black suit, keep neon katana",                      # 服装替换
        "Change rain to heavy snow, add frost on shoulders",                                   # 天气调整
        "Convert the scene to steampunk style, brass mechanical details",                      # 风格迁移
        "Add a drone hovering above him, scanning with red laser",                              # 科技元素
        "Make him half-robot with visible circuitry, biomechanical"                            # 角色改造
    ],
    # 原图8: "A bee's eye view of a sunflower pollen field"
    [
        "Show the sunflowers from human eye level, keep pollen details",                       # 视角切换
        "Change the scene to autumn, withered flowers, golden leaves",                          # 季节变化
        "Zoom in to show pollen grains under electron microscope",                              # 显微细节
        "Add giant honey crystals growing between flowers",                                     # 超现实元素
        "Convert to ultraviolet insect vision, fluorescent colors"                             # 特殊视觉效果
    ],
    # 原图9: "Blue and gold fluid art resembling a galaxy"
    [
        "Invert colors to deep red and black, keep fluid dynamics",                            # 色彩反转
        "Transform the fluid into sharp geometric fractals",                                   # 形态重构
        "Add motion lines to simulate swirling galaxy animation",                              # 动态化
        "Embed constellations and zodiac symbols in the fluid",                                # 元素融合
        "Make the fluid solidify into crystalline structures"                                  # 物理状态变化
    ],
    # 原图10: "A medieval knight charging on a motorcycle"
    [
        "Replace the motorcycle with an armored war horse, keep knight",                       # 逻辑合理化
        "Make the knight wear nano-armor, futuristic plasma lance",                            # 时代迁移
        "Add a dragon attacking the knight, dramatic fire and smoke",                          # 冲突增强
        "Convert to sketch outline drawing, monochrome ink style",                             # 线稿化
        "Make the scene a toy diorama with miniature props"                                    # 比例变形
    ]
]

for i in range(10):
  print(f"第{i}张照片准备好")
  prompt = prompt_list[i]
  image = pipe(
    prompt,
    height=1024,
    width=1024,
    guidance_scale=3.5,
    num_inference_steps=50,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(seed)
  ).images[0]
  image.save(f"output/kontext-dev/kontext-dev_gen{i}.jpg")
  print(f"第{i}张照片已生成&保存")
  for j in range(5):
    edit_prompt = edit_prompt_list[j]
    image = pipe(
      prompt,
      height=1024,
      width=1024,
      guidance_scale=3.5,
      num_inference_steps=50,
      max_sequence_length=512,
      generator=torch.Generator("cpu").manual_seed(seed)
    ).images[0]
    image.save(f"output/kontext-dev/kontext-dev_edit{i}-{j}.jpg")
    print(f"第{i}-{j}张照片已编辑&保存")

print("all done")
